server/models/repositoryModel.js

```diff
@@ -55,24 +53,44 @@ repositorySchema.statics.get = async () => {
     return await RepositoryModel.find({})
 }
 
-repositorySchema.statics.addFeature = async (taskInput, user) => {
-    if (!user || (!userHasRole(user, ROLE_NEGOTIATOR) && !userHasRole(user, ROLE_ESTIMATOR)))
-        throw new AppError('Only user with any of the roles [' + ROLE_ESTIMATOR + "," + ROLE_NEGOTIATOR + "] can add task to repository", INVALID_USER, HTTP_BAD_REQUEST)
+repositorySchema.statics.addFeature = async (featureInput, user) => {
+    if (!user || (!userHasRole(user, SC.ROLE_NEGOTIATOR) && !userHasRole(user, SC.ROLE_ESTIMATOR)))
+        throw new AppError('Only user with any of the roles [' + SC.ROLE_ESTIMATOR + "," + SC.ROLE_NEGOTIATOR + "] can add feature to repository", EC.INVALID_USER, HTTP_BAD_REQUEST)
 
-    validate(taskInput, repositoryAddTaskStruct)
+    validate(featureInput, repositoryAddTaskStruct)
 
-    const estimation = await EstimationModel.findById(taskInput.estimation._id)
+    const estimation = await EstimationModel.findById(featureInput.estimation._id)
     if (!estimation)
-        throw new AppError('Estimation not found', NOT_FOUND, HTTP_BAD_REQUEST)
+        throw new AppError('Estimation not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
 
-    taskInput.status = STATUS_PENDING
-    taskInput.isFeature = true
-    taskInput.isPartOfEstimation = true
-    taskInput.type = TYPE_DEVELOPMENT
-    taskInput.foundationTask = false
-    taskInput.hasHistory = false
-    taskInput.createdBy = user
-    return await RepositoryModel.create(taskInput)
+    featureInput.status = SC.STATUS_PENDING
+    featureInput.isFeature = true
+    featureInput.isPartOfEstimation = true
+    featureInput.type = SC.TYPE_DEVELOPMENT
+    featureInput.foundationTask = false
+    featureInput.hasHistory = false
+    featureInput.createdBy = user
+    return await RepositoryModel.create(featureInput)
+}
+
+repositorySchema.statics.updateFeature = async (featureInput, user) => {
+    if (!user || (!userHasRole(user, SC.ROLE_NEGOTIATOR) && !userHasRole(user, SC.ROLE_ESTIMATOR)))
+        throw new AppError('Only user with any of the roles [' + SC.ROLE_ESTIMATOR + "," + SC.ROLE_NEGOTIATOR + "] can update feature to repository", EC.INVALID_USER, EC.HTTP_BAD_REQUEST)
+    if(!featureInput.repo && !featureInput.repo._id)
+        throw new AppError('Repository not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    const repository = await RepositoryModel.findById(featureInput.repo._id)
+
+    if(_.includes([SC.STATUS_APPROVED], repository.status))
+        throw new AppError("Repository has feature status as ["+repository.status+"]. Estimator can only update feature into those estimations where status is in [" + SC.STATUS_ESTIMATION_REQUESTED + ", " + SC.STATUS_CHANGE_REQUESTED + "]", EC.INVALID_OPERATION, EC.HTTP_BAD_REQUEST)
+
```

Comment:
Features from repository can only be updated if that feature was added by same estimation